### DISCUSSION TITLE
How to Retrieve the JID of a User Reacting to a Message in a Group with whatsmeow

### DISCUSSION THREAD

**Discussion by andersonribeir0** [tulir/whatsmeow#485] [WhatsApp protocol Q&A] [answered]
Hello everyone,

I'm currently working with the whatsmeow library and I'm facing a challenge in identifying the JID of a user who reacts to a message within a WhatsApp group. From what I've understood, the ReactionMessage event provides several fields, but I'm particularly interested in determining the unique identifier of the individual who added the reaction.

The fields such as RemoteJID seem to give me the JID of the group, and Participant appears to provide the JID of the original message sender. However, neither of these fields point me to the JID of the user who reacted.

Here's what I'm observing in the ReactionMessage structure:

```
state         protoimpl.MessageState
sizeCache     protoimpl.SizeCache
unknownFields protoimpl.UnknownFields

Key               *MessageKey `protobuf:"bytes,1,opt,name=key" json:"key,omitempty"`
Text              *string     `protobuf:"bytes,2,opt,name=text" json:"text,omitempty"`
GroupingKey       *string     `protobuf:"bytes,3,opt,name=groupingKey" json:"groupingKey,omitempty"`
SenderTimestampMs *int64      `protobuf:"varint,4,opt,name=senderTimestampMs" json:"senderTimestampMs,omitempty"`
```

Could someone please guide me on where to find the JID of the user who added the reaction? Is there a specific field or part of the event structure that I should be looking at?

Any assistance or pointers in the right direction would be greatly appreciated.

Thank you for your help!

**Comment by tulir** [comment, marked as answer]
Incoming message info is in the [`MessageInfo` struct](https://pkg.go.dev/go.mau.fi/whatsmeow/types#MessageInfo), which can be found in the `Info` field of [`*events.Message`](https://pkg.go.dev/go.mau.fi/whatsmeow/types/events#Message)
